api/v1alpha1/edgedeployment_types.go

```diff
@@ -41,18 +42,8 @@ const (
 	PodDeploymentType EdgeDeploymentType = "pod"
 )
 
-type EdgeDeploymentPhase string
-
-const (
-	Deploying EdgeDeploymentPhase = "Deploying"
-	Running   EdgeDeploymentPhase = "Running"
-	Exited    EdgeDeploymentPhase = "Exited"
-)
-
 // EdgeDeploymentStatus defines the observed state of EdgeDeployment
 type EdgeDeploymentStatus struct {
```

Comment:
After this change deployment may run on one or more devices why not to use something similar to `deployment` status:

```yaml
status:
  availableReplicas: 2
  conditions:
  - lastTransitionTime: 2016-10-04T12:25:39Z
    lastUpdateTime: 2016-10-04T12:25:39Z
    message: Replica set "nginx-deployment-4262182780" is progressing.
    reason: ReplicaSetUpdated
    status: "True"
    type: Progressing
  - lastTransitionTime: 2016-10-04T12:25:42Z
    lastUpdateTime: 2016-10-04T12:25:42Z
    message: Deployment has minimum availability.
    reason: MinimumReplicasAvailable
    status: "True"
    type: Available
  - lastTransitionTime: 2016-10-04T12:25:39Z
    lastUpdateTime: 2016-10-04T12:25:39Z
    message: 'Error creating: pods "nginx-deployment-4262182780-" is forbidden: exceeded quota:
      object-counts, requested: pods=1, used: pods=3, limited: pods=2'
    reason: FailedCreate
    status: "True"
    type: ReplicaFailure
  observedGeneration: 3
  replicas: 2
  unavailableReplicas: 2
```

Reply:
IMO so complex status is not needed at this stage. 
I thought about keeping simple deployment status (as it was before that change) per device in the EdgeDeployment CR and decided to put it in the EdgeDevice because the reference from device to deployments was needed. Now I have second thoughts about it - labels on the EdgeDevice serve the purpose and maybe keeping per device  status in EdgeDeployment is a better option. WDYT about moving it back to the EdgeDeployment CR?

Reply:
sounds like a good idea

Reply:
I remembered another reason for moving the deployment statuses to the EdgeDevice CR - potentially the list of devices in the EdgeDeployment CR could be massive in contrast to number of deployments per device, which should be rather small. 

I remember from the discussions that having long lists of devices for precise scheduling is not desired and here we would have additional data besides the IDs making it even bigger.

Reply:
As per our discussion let's keep it as it is and adjust when user feedback instructs us to redesign.